source/Test.cpp

```diff
@@ -84,6 +107,70 @@ Test::TestStep::TestStep(Type stepType) : stepType(stepType)
 
 
 
+void Test::TestStep::LoadInput(const DataNode &node)
+{
+	for(const DataNode &child : node)
+	{
+		if(child.Token(0) == "key")
+		{
+			for(int i=1; i < child.Size(); ++i)
+				inputKeys.insert(child.Token(i));
+			
+			for(const DataNode &grand: child){
+				if(grand.Token(0) == "shift")
+					modKeys |= KMOD_SHIFT;
+				else if(grand.Token(0) == "alt")
+					modKeys |=  KMOD_ALT;
+				else if(grand.Token(0) == "control")
+					modKeys |= KMOD_CTRL;
+				else
+					grand.PrintTrace("Warning: Unknown keyword in \"input\" \"key\" section:");
+			}
+		}
+		else if(child.Token(0) == "pointer")
+		{
+			for(const DataNode &grand: child){
+				if(grand.Token(0) == "X")
+				{
+					if(grand.Size() < 2)
+						grand.PrintTrace("Warning: Pointer X axis input without coordinate:");
+					else{
+						XValue = grand.Value(1);
+					}
+				}
+				else if(grand.Token(0) == "Y")
+				{
+					if(grand.Size() < 2)
+						grand.PrintTrace("Warning: Pointer Y axis input without coordinate:");
+					else{
+						YValue = grand.Value(1);
+					}
+				}
+				else if(grand.Token(0) == "click")
+					for(int i=1; i < grand.Size(); ++i)
+					{
+						if(grand.Token(i) == "left")
+							clickLeft = true;
+						else if(grand.Token(i) == "right")
+							clickRight = true;
+						else if(grand.Token(i) == "middle")
+							clickMiddle = true;
+						else
+							grand.PrintTrace("Warning: Unknown click/button:");
```

Comment:
The trace will print the entire line, which per above comment may contain multiple click buttons. If multiple are allowed, include the unknown token to improve the message.